src/nvim/screen.c

```diff
@@ -4297,9 +4319,15 @@ void draw_virt_text(buf_T *buf, int col_off, int *end_col, int max_col)
     if (item->win_col < 0) {
       continue;
     }
-
-    int col = draw_virt_text_item(buf, item->win_col, item->decor.virt_text,
-                                  item->decor.hl_mode, max_col, item->win_col-col_off);
+    int col;
+    if (item->decor.ui_watched) {
+      // don't draw watched decors, let ui handle that.
+      col = item->win_col;
+      win_extmarks_add(item->ns_id, item->mark_id, win_row, col);
+    } else {
```

Comment:
then, this can be disentangled. so not an `else` anymore, just `if (kv_size(item->decor.virt_text))`

Reply:
I read your comment multiple times to make sure I understand: allow a mark to be both ui watched and drawn on grid?

Reply:
yes, otherwise the attribute `virt_text` is just ignored, and there is no need to do so any longer. if a plugin doesn't want this, they can just set `ui_watched` now.

Reply:
got it. done in https://github.com/neovim/neovim/pull/15674/commits/c6c7105bf0cbe1ea6a524cf4c1e41d1bdb062cea